File: Data Structures & Algorithms/longest-substring-without-duplicates/submission-4.cpp
class Solution {
public:
    int lengthOfLongestSubstring(string s) {
        unordered_map<char, int> mp;
        int ans = 0;
        int i = 0, j = 0;
        while(j < s.size() && i < s.size()){
            if(mp[s[j]] > 0){
                i = max(mp[s[j]], i);
            }
            //
                ans = max((j - i + 1), ans);
                cout<<i<<" "<<j<<"\n";
            //}
            mp[s[j]] = j + 1;
            j++;
        }
        cout<<s[20];
        return ans;
    }
};
//s="pwwkew"

File: Data Structures & Algorithms/longest-consecutive-sequence/submission-4.java
class Solution {
    public int longestConsecutive(int[] nums) {
        if(nums.length == 0){
            return 0;
        }
        HashSet<Integer> hs = new HashSet<>();
        for(int i:nums) {
            hs.add(i);
        }
        int ans = 1;
        for(int j:nums) {
            if(hs.contains(j+1) && !hs.contains(j-1)){
                int itr = j;
                int k  = 1;
                while(hs.contains(itr+1)) {
                    k++;
                    itr++;
                }
                ans = Math.max(k,ans);
            }
        }
        return ans;
    }
}
